src/core/database/database.service.ts

```diff
@@ -555,34 +555,46 @@ export class DatabaseService {
   }: {
     object: TObject;
   }) {
+    const deletedAt = DateTime.local();
     const query = this.db
       .query()
       .match([
         node('baseNode', { id: object.id }),
         // in this case we want to set Deleted_ labels for all properties
-        // including those with active = false
-        relation('out', 'propRel'),
+        // including active = false
+        // deleteProperties does this, but deletes from before that was changed only prefixed
+        // unique property labels
+        relation('out', ''),
         node('propertyNode', 'Property'),
       ])
       // Mark any parent base node relationships (pointing to the base node) as active = false.
       .optionalMatch([
         node('baseNode'),
-        relation('in', 'rel'),
+        relation('in', 'baseNodeRel'),
         node('', 'BaseNode'),
       ])
       .setValues({
-        'rel.active': false,
-        'propRel.active': false,
+        'baseNode.deletedAt': deletedAt,
+        'baseNodeRel.active': false,
       })
-      // after setting propRel.active false, we need to distinct propertyNode (not sure why)
-      // combine baseNode with propertyNodes to set Deleted_ labels on all of them
-      .with('[baseNode] + collect(distinct propertyNode) as nodeList')
-      // yielding a node from this procedure is necessary I believe, but not used in the rest of the query
-      // they're aliased to avoid colliding with the unwound node var
+      // if we set anything on property nodes or property relationships in the query above
+      // (as was done before, but now now)
+      // we need to distinct propertyNode since each resulting row has both propertyNode and baseNode I believe
+      .with('[baseNode] + collect(propertyNode) as nodeList')
```

Comment:
You say we need `distinct` but it's been removed from the query...?

Reply:
It's only needed if we do anything to property nodes/relationships (as the query was previously doing). As the query is written now distinct does nothing. Maybe the comment is unnecessary, but I wanted to prevent bugs if we modify it down the road.

Reply:
Can you update the comment to reflect the current thought?

Reply:
Yep. Should be clearer now.

Reply:
> we need to distinct propertyNode to avoid collecting and labeling each propertyNode more than once

Then why aren't we using `distinct`?

Reply:
It needs to be understood with the first line of the comment.

Reply:
Ah I see.